Standardize case of symbols in libsupport

ddn0 opened the pull request:
There is a mix of styles when writing a symbol that is an
initialism/acronym: RDG, Json, Http. Instead consistently apply
consecutive capitals for these cases.

tylershunt commented on libsupport/src/FileSystem.cpp:
```diff
@@ -74,12 +74,22 @@ galois::Result<std::string> galois::NewPath(const std::string& dir,
 }
 
+// TODO: These functions do not properly deal with non-canonical path
+//  names, e.g., multiple '/' in a row
```
Should probably be `TODO(witchel)` or maybe `TODO(scober)`

tylershunt commented on libsupport/src/FileSystem.cpp:
```diff
@@ -74,12 +74,22 @@ galois::Result<std::string> galois::NewPath(const std::string& dir,
+    return ErrorCode::InvalidArgument;
+  }
+  return path.substr(0, last_slash);
+}
```
The behavior here is to return `some/long/path` for an input like `/some/long/path/` that end in  `kSepChar`? I only ask because `dirname` would return `some/long` in that case.

witchel replied:
Sure, I can do this.  I'll point out that ExtractFileName would return the empty string if you send it  `/some/long/path/`

tylershunt replied:
No real preference just wondering about the semantics. `basename /some/long/path/` returns `path` so maybe we should change there too?

witchel replied:
This code is ugly.  Maybe there is a better way?

tylershunt commented on libtsuba/include/tsuba/FaultTest.h:
```diff
@@ -23,7 +23,8 @@ GALOIS_EXPORT void FaultTestInit(FaultMode mode         = FaultMode::None,
 // PullThePlug (virtually) Compile this out if NDEBUG?
-void PtP(FaultSensitivity sensitivity = FaultSensitivity::Normal);
+void PtP(const char* file, int line,
+         FaultSensitivity sensitivity = FaultSensitivity::Normal);
```
These are getting verbose; we might think about adding a macro like:
`#DEFINE TSUBA_PTP(...) ::tsuba::internal::PtP(__FILE__, __LINE__, ##__VA_ARGS__)`

witchel replied:
I agree it is getting ugly.  For some reason, I thought we had a "ban" on cpp, but I'm happy to turn this into a macro.

tylershunt replied:
Yeah I think we want to avoid cpp macros in general, but for things involving  `__FILE__` and `__LINE__` I don't know of anything better.

ddn0 committed:
Standardize case of symbols in libsupport

There is a mix of styles when writing a symbol that is an
initialism/acronym: RDG, Json, Http. Instead consistently apply
consecutive capitals for these cases.
